### DISCUSSION TITLE
From VS 2019 to VC Code PowerShell stops working

### DISCUSSION THREAD

**Discussion by mezamoe** [MahApps/MahApps.Metro#4103] [Q&A] [answered]
Hello, before posting in the Issues section I'd like to check with you whether what I'm doing is correct. I built a WPF App in VS 2019 as a WPF .NET Framework project and it works. I really like the result and now I want to translate it into ps1 file. I followed what adviced in this link:

http://vcloud-lab.com/entries/powershell/part-3-powershell-wpf-mahapps-metro-theme-step-by-step 

but my ps1 does not work. The only difference is that I'm using Microsoft.Xaml.Behaviors.dll rather than System.Windows.Interactivity.dll.
I also copied the file from that URL and it throws the same error message I get on my ps1 script:

Exception calling "Load" with "1" argument(s): "Cannot create unknown type '{clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro}MetroWindow'."

Here's my xaml:

```
<Controls:MetroWindow 
        xmlns:Controls="clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        Title="Test" Height="325" Width="735" WindowStartupLocation="CenterScreen" ResizeMode="NoResize" Topmost="True" WindowStyle="ThreeDBorderWindow" FontFamily="Calibri" Foreground="Black" Background="White" BorderBrush="#FFF9FAFB" BorderThickness="0,0,0,0">

    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Fonts.xaml" />
                <!-- Theme setting -->
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Themes/Light.Blue.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>

    </Application.Resources>
 
    <Grid >

    </Grid>
</Controls:MetroWindow>
```

Here the ps1 file. When it get to the Try statement it throws the error.

```
# Load the Assemblies 
[void][System.Reflection.Assembly]::LoadWithPartialName('PresentationCore')
[void][System.Reflection.Assembly]::LoadWithPartialName('PresentationFramework')
[void][System.Reflection.Assembly]::LoadWithPartialName('WindowsBase')
[void][System.Reflection.Assembly]::LoadWithPartialName('system.windows.forms')
[void][System.Reflection.Assembly]::LoadWithPartialName('MahApps.Metro.dll') 
[void][System.Reflection.Assembly]::LoadWithPartialName('ControlzEx.dll') 
[void][System.Reflection.Assembly]::LoadWithPartialName('Microsoft.Xaml.Behaviors.dll') 

[xml]$xamlWindow  = Get-Content ".\app.xaml"
$xamlReader  = (New-Object System.Xml.XmlNodeReader $xamlWindow)
try {
    $MainForm=[Windows.Markup.XamlReader]::Load( $xamlReader )
} catch {
    Write-Host "Unable to load Windows.Markup.XamlReader. Please check the XAML code entered."
    exit
}
$xamlWindow.SelectNodes("//*[@Name]") | ForEach-Object{Set-Variable -Name ($_.Name) -Value $MainForm.FindName($_.Name)}
$MainForm.ShowDialog() | out-null
```

**Comment by timunie** [comment]
Hi @mezamoe 

I have no idea about PowerShell but what I can tell from the XAML you posted is, that you need to change `Application.Resources` to `Controls:MetroWindow.Resources`. This is also written in the link you posted. At least the window now loads for me in a "normal" WPF app. 

```xaml
<Controls:MetroWindow 
        xmlns:Controls="clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        Title="Test" Height="325" Width="735" WindowStartupLocation="CenterScreen" ResizeMode="NoResize" Topmost="True" WindowStyle="ThreeDBorderWindow" FontFamily="Calibri" Foreground="Black" Background="White" BorderBrush="#FFF9FAFB" BorderThickness="0,0,0,0">

    <Controls:MetroWindow.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml" />
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Fonts.xaml" />
                <!-- Theme setting -->
                <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Themes/Light.Blue.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>

    </Controls:MetroWindow.Resources>
 
    <Grid >

    </Grid>
</Controls:MetroWindow>
```

I hope this helps.  

Happy coding
Tim

**Comment by mezamoe** [comment]
Hi Tim,
thanks for your help! The link I posted does not mention that.. Controls:MetroWindow  is the tag for the main window not the for the resources to be used inside it. In fact he calls that tag as Window.Resources. It is puzzling that it works for you...  I'm more confused than before..  :)

**Reply by timunie** [reply]
Ok he uses `Window.Resources` which works because `MetroWindow` derives from `Window`, but not from `Application`.

**Reply by mezamoe** [reply]
Thanks for clarifying! The error remains the same, it's really weird... if it works for you there must be something with my setup then..

```
Exception calling "Load" with "1" argument(s): "Cannot create unknown type '{clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro}MetroWindow'."
At line:1 char:1
+ $MainForm=[Windows.Markup.XamlReader]::Load( $xamlReader )
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : XamlParseException
```

**Reply by punker76** [reply]
@mezamoe It seems like that the MahApps, ControlzEx and Behaviors assemblies are not loaded. Where is your folder from which these assemblies should beloaded? I have here a test repo which loads the assemblies form a local folder https://github.com/punker76/dialog_validation_test Hopefully this helps you.

**Reply by mezamoe** [reply]
thanks Punker76! I will check in a bit and revert... all is leading to a misconfig on my side

**Reply by mezamoe** [reply]
I cloned that repo but I get the same error! If there is anything wrong with my config I can't figure it out! At first line 5 could not retrieve the dll files and I added -recurse, now it does. But the error message even here is still the same..!!! 

```
Exception calling "Load" with "1" argument(s): "Cannot create unknown type '{clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro}MetroWindow'."
At C:\Users\xxxxxx\gitS\dialog_validation_test\Testing_GUI.ps1:19 char:1
+ $Form=[Windows.Markup.XamlReader]::Load($Reader)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : XamlParseException
```

**Reply by mezamoe** [reply]
and yes there was something wrong... the tagging!

the namespace was missing the correct identifier:

xmlns:**mah**="clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro" 

Thanks both for your time!!

**Comment by mezamoe** [comment, marked as answer]
and yes there was something wrong... the tagging!

the namespace was missing the correct identifier:

xmlns:mah="clr-namespace:MahApps.Metro.Controls;assembly=MahApps.Metro"
